Mathlib/Geometry/Manifold/Instances/Real.lean

```diff
@@ -245,6 +245,33 @@ def IccLeftChart (x y : ℝ) [h : Fact (x < y)] :
     have B : Continuous fun z : EuclideanSpace ℝ (Fin 1) => z 0 := continuous_apply 0
     exact (A.comp B).comp continuous_subtype_val
 
+variable {x y : ℝ} [hxy : Fact (x < y)]
+
+/-- The endpoint `x ∈ Icc x y`, as a point in `Icc x y` (assuming `x < y`). -/
+abbrev IccManifold.left : Icc x y := ⟨x, ⟨le_refl x, by have := hxy.out; linarith⟩⟩
+
+/-- The endpoint `y ∈ Icc x y`, as a point in `Icc x y` (assuming `x < y`). -/
+abbrev IccManifold.right : Icc x y := ⟨y, ⟨by have := hxy.out; linarith, le_refl y⟩⟩
+
+open IccManifold (left right)
+
+lemma IccLeftChart_extend_left_eq : ((IccLeftChart x y).extend (𝓡∂ 1)) left = 0 := by
+  calc ((IccLeftChart x y).extend (𝓡∂ 1)) left
+    _ = (𝓡∂ 1) ⟨fun _ ↦ 0, by norm_num⟩ := by norm_num [IccLeftChart]
+    _ = 0 := rfl
+
+lemma IccLeftChart_extend_interior_pos {p : Set.Icc x y} (hp : x < p.val ∧ p.val < y) :
+    (((IccLeftChart x y).extend (𝓡∂ 1)) p) 0 > 0 := by
+  set lhs := (IccLeftChart x y).extend (𝓡∂ 1) p
+  have : lhs 0 = p.val - x := rfl
+  rw [this]
+  norm_num [hp.1]
+
+lemma IccLeftChart_extend_left_mem_frontier :
```

Comment:
I'm open to the argument that this lemma (and its right counterpart) is sufficiently trivial to be inlined.